Application/EdFi.Ods.Tests/EdFi.Ods.Api/Middleware/YearSpecificContextTests.cs

```diff
@@ -0,0 +1,103 @@
+            _schoolYearContextProvider = A.Fake<ISchoolYearContextProvider>();
+
+            _httpContext = HttpContextHelper.GetActionContext().HttpContext;
+        }
+
+        [Test]
+        public async Task Should_Not_Parse_School_Year_When_An_Array()
+        {
+            _httpContext.Request.RouteValues.Add("schoolYearFromRoute", "2010, 2011");
+
+            var sut = new SchoolYearRouteContextMiddleware(_schoolYearContextProvider);
+
+            await sut.InvokeAsync(_httpContext, context => Task.CompletedTask);
+
+            A.CallTo(() => _schoolYearContextProvider.SetSchoolYear(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public async Task Should_Not_Parse_InValid_School_Year()
+        {
+            _httpContext.Request.RouteValues.Add("schoolYearFromRoute", "abc");
+
+            var sut = new SchoolYearRouteContextMiddleware(_schoolYearContextProvider);
+
+            await sut.InvokeAsync(_httpContext, context => Task.CompletedTask);
+
+            A.CallTo(() => _schoolYearContextProvider.SetSchoolYear(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public async Task Should_Not_Parse_A_Null_School_Year()
+        {
+            _httpContext.Request.RouteValues.Add("schoolYearFromRoute", null);
+
+            var sut = new SchoolYearRouteContextMiddleware(_schoolYearContextProvider);
+
+            await sut.InvokeAsync(_httpContext, context => Task.CompletedTask);
+
+            A.CallTo(() => _schoolYearContextProvider.SetSchoolYear(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public async Task Should_Not_Parse_An_Empty_School_Year()
+        {
+            _httpContext.Request.RouteValues.Add("schoolYearFromRoute", null);
+
+            var sut = new SchoolYearRouteContextMiddleware(_schoolYearContextProvider);
+
+            await sut.InvokeAsync(_httpContext, context => Task.CompletedTask);
+
+            A.CallTo(() => _schoolYearContextProvider.SetSchoolYear(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public async Task Should_Not_Parse_Whitespace_School_Year()
+        {
+            _httpContext.Request.RouteValues.Add("schoolYearFromRoute", " ");
+
+            var sut = new SchoolYearRouteContextMiddleware(_schoolYearContextProvider);
+
+            await sut.InvokeAsync(_httpContext, context => Task.CompletedTask);
+
+            A.CallTo(() => _schoolYearContextProvider.SetSchoolYear(A<int>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public async Task Should_Process_A_Valid_School_Year()
+        {
+            var year = DateTime.Now.Year;
+            _httpContext.Request.RouteValues.Add("schoolYearFromRoute", year);
+
+            var sut = new SchoolYearRouteContextMiddleware(_schoolYearContextProvider);
+
+            await sut.InvokeAsync(_httpContext, context => Task.CompletedTask);
+
+            A.CallTo(() => _schoolYearContextProvider.SetSchoolYear(year)).MustHaveHappened();
```

Comment:
Same here, be explicit that it should set school year exactly once.

Reply:
done